src/mqt/bench/targets/gatesets/rigetti.py

```diff
@@ -0,0 +1,173 @@
+from qiskit.circuit import Gate, Parameter, SessionEquivalenceLibrary
+from qiskit.circuit.library import CXGate, iSwapGate
+from qiskit.circuit.library.standard_gates import RZGate, UGate
+
+
+def get_rigetti_gateset() -> list[str]:
+    """Returns the basis gates of the Rigetti gateset."""
+    add_rigetti_equivalences()
+    return ["fixed_angle_rx_pi", "fixed_angle_rx_pi_by_2", "fixed_angle_rx_pi_by_minus_2", "rz", "iswap", "measure"]
+
+
+class RXPiGate(Gate):  # type: ignore[misc]
+    r"""Single-qubit RX(π) gate.
+
+    **Circuit symbol:**
+    .. parsed-literal::
+             ┌─────┐
+        q_0: ┤ RX(π)├
+             └─────┘
+
+    **Matrix representation:**
+
+    .. math::
+
+        RX(π) =
+            \begin{pmatrix}
+                0 & -i \\
+                -i & 0
+            \end{pmatrix}
+    """
+
+    def __init__(self, label: str | None = None) -> None:
+        """Create RX(π) gate."""
+        super().__init__("fixed_angle_rx_pi", 1, [np.pi], label=label)
+
+    def _define(self) -> None:
+        """Define RX(π) gate using standard RX."""
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.rx(np.pi, 0)
+        self.definition = qc
+
+
+class RXPiOver2Gate(Gate):  # type: ignore[misc]
+    r"""Single-qubit RX(π/2) gate.
+
+    **Circuit symbol:**
+    .. parsed-literal::
+             ┌──────────┐
+        q_0: ┤ RX(π/2)  ├
+             └──────────┘
+
+    **Matrix representation:**
+
+    .. math::
+
+        RX(π/2) =
+            \begin{pmatrix}
+                \cos(π/4) & -i \sin(π/4) \\
+                -i \sin(π/4) & \cos(π/4)
+            \end{pmatrix}
+    """
+
+    def __init__(self, label: str | None = None) -> None:
+        """Create RX(π/2) gate."""
+        super().__init__("fixed_angle_rx_pi_by_2", 1, [np.pi / 2], label=label)
+
+    def _define(self) -> None:
+        """Define RX(π/2) gate using standard RX."""
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.rx(np.pi / 2, 0)
+        self.definition = qc
+
+
+class RXMinusPiOver2Gate(Gate):  # type: ignore[misc]
+    r"""Single-qubit RX(-π/2) gate.
+
+    **Circuit symbol:**
+    .. parsed-literal::
+             ┌────────────┐
+        q_0: ┤ RX(-π/2)   ├
+             └────────────┘
+
+    **Matrix representation:**
+
+    .. math::
+
+        RX(-π/2) =
+            \begin{pmatrix}
+                \cos(π/4) & i \sin(π/4) \\
+                i \sin(π/4) & \cos(π/4)
+            \end{pmatrix}
+    """
+
+    def __init__(self, label: str | None = None) -> None:
+        """Create RX(-π/2) gate."""
+        super().__init__("fixed_angle_rx_pi_by_minus_2", 1, [-np.pi / 2], label=label)
+
+    def _define(self) -> None:
+        """Define RX(-π/2) gate using standard RX."""
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.rx(-np.pi / 2, 0)
+        self.definition = qc
+
+
+def u_gate_equivalence() -> None:
```

Comment:
I think it could be worth to add a couple more equivalences here that take fewer gates.
Equivalences for the real (parametrized) RX gate come to mind.
Pretty sure there are others.

Reply:
I added one for that, although it still has 5 gates (but at least without parameters - not sure how much it improves though).